craft_parts/plugins/rust_plugin.py

```diff
@@ -111,32 +118,84 @@ def get_build_snaps(self) -> Set[str]:
     @override
     def get_build_packages(self) -> Set[str]:
         """Return a set of required packages to install in the build environment."""
-        return {"gcc", "git"}
+        return {"curl", "gcc", "git", "pkg-config", "findutils"}
+
+    def _check_rustup(self) -> bool:
+        try:
+            rustup_version = subprocess.check_output(["rustup", "--version"])
+            return "rustup" in rustup_version.decode("utf-8")
+        except (subprocess.CalledProcessError, FileNotFoundError):
+            return False
+
+    def _get_setup_rustup(self, channel: str) -> List[str]:
+        return [
+            f"""\
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | \
+sh -s -- -y --no-modify-path --profile=minimal --default-toolchain {channel}
+"""
+        ]
 
     @override
     def get_build_environment(self) -> Dict[str, str]:
         """Return a dictionary with the environment to use in the build step."""
-        return {"PATH": "${HOME}/.cargo/bin:${PATH}"}
+        return {
+            "PATH": "${HOME}/.cargo/bin:${PATH}",
+        }
 
     @override
     def get_build_commands(self) -> List[str]:
         """Return a list of commands to run during the build step."""
         options = cast(RustPluginProperties, self._options)
 
-        install_command_list = [
-            "cargo",
-            "install",
-            "--locked",
-            "--path",
-            options.rust_path[0],  # TODO: support more than 1 path
-            "--root",
-            '"${CRAFT_PART_INSTALL}"',
-            "--force",
-        ]
+        rust_install_cmd: List[str] = []
+        rust_build_cmd: List[str] = []
+        config_cmd: List[str] = []
+
+        if options.rust_channel == "none":
+            rust_install_cmd = []
```

Comment:
That's a good point. However, I do think if the user chose the update channel instead of a fixed version, they should expect the version to be updated each time the upstream releases a new compiler version.

Reply:
That's a good point. In that case we probably do want to run `rustup update` since I would guess it would be less surprising to get a new version of rust than to come back to a project you haven't touched in a month and get an old version of rust.

Reply:
> That's a good point. In that case we probably do want to run `rustup update` since I would guess it would be less surprising to get a new version of rust than to come back to a project you haven't touched in a month and get an old version of rust.

For Rust, if you are using the `stable` channel (which is the default), it is improbable that the newer compiler would break your project. Even if it does, the Rust compiler usually will provide an option to fix your code automatically.

Reply:
Exactly — and if you're using the `nightly` channel you can (theoretically at least) expect your code to break randomly in CI unless you've set up a `rust-toolchain.toml`.

But in a case like [the alacritty snap](https://github.com/snapcrafters/alacritty/blob/1f478744ea69f1095c5399e65db2888b9d598ebe/snap/snapcraft.yaml), if you have `rust-channel: stable` update the `source-tag`, you don't want it to tell you that rust is out of date and to have to `snapcraft clean` to get it to work — instead, re-pulling should include updating rust.